zebra-chain/src/parallel/tree.rs

```diff
@@ -132,12 +138,44 @@ impl NoteCommitmentTrees {
         if let Some(sprout_result) = sprout_result {
             self.sprout = sprout_result?;
         }
+
         if let Some(sapling_result) = sapling_result {
-            self.sapling = sapling_result?;
-        }
+            let (sapling, subtree_root) = sapling_result?;
+            self.sapling = sapling;
+            self.sapling_subtree =
+                subtree_root.map(|(subtree_root, starts_subtree, subtree_index)| {
+                    sapling::tree::NoteCommitmentSubtree {
+                        index: subtree_index,
+                        end: height,
+                        next_start: if starts_subtree {
+                            height
+                        } else {
+                            height.next()
+                        },
+                        subtree_root,
+                    }
+                    .into()
+                });
+        };
+
         if let Some(orchard_result) = orchard_result {
-            self.orchard = orchard_result?;
-        }
+            let (orchard, subtree_root) = orchard_result?;
+            self.orchard = orchard;
+            self.orchard_subtree =
+                subtree_root.map(|(subtree_root, starts_subtree, subtree_index)| {
+                    orchard::tree::NoteCommitmentSubtree {
+                        index: subtree_index,
+                        end: height,
+                        next_start: if starts_subtree {
+                            height
+                        } else {
+                            height.next()
```

Comment:
This should be the next block with notes, not always the next block height.

Reply:
Converted to a flag in https://github.com/ZcashFoundation/zebra/pull/7334/commits/5dd93ec4f0cd3eb3da9bac87b5bf751e72cfc780.